src/server/schema/track.schema.ts

```diff
@@ -34,7 +34,9 @@ export const trackShareSchema = z.object({
   url: z.string().url().trim().nonempty(),
 });
 
-export type TrackEventInput = z.infer<typeof trackEventSchema>;
-export const trackEventSchema = z.object({
-  type: z.enum(EventType),
+export type TrackActionInput = z.infer<typeof trackActionSchema>;
+export const trackActionSchema = z.object({
+  type: z.enum(ActionType),
+  // TODO.tracking: better typing
+  details: z.object({}).passthrough().optional(),
```

Comment:
You can always use discriminated unions for this I'd say. I did that for the paymentIntents (With a fallback to passthrough). useful to add **SOME** typing without the need to go overboard